### PR TITLE
add GetMaps, uses redis pipeline to get multiple key maps

### PR DESCRIPTION
**Pull request by louiseGrandjonc**
in one redis call

### CODE REVIEW

File: dummy.go [hunk 1]
```diff
@@ -18,6 +18,11 @@ func (s DummyStore) GetMap(key string) (map[string]interface{}, error) {
 	return nil, nil
 }
 
+// GetMaps returns maps for the given keys.
+func (s *DummyStore) GetMaps(keys []string) (map[string]map[string]interface{}, error) {
```
Comment (gillesfabio) [inline dummy.go]: `func (DummyStore)` is enough, as we return nil.

File: dummy.go [hunk 2]
```diff
@@ -18,6 +18,11 @@ func (s DummyStore) GetMap(key string) (map[string]interface{}, error) {
...
+	return nil, nil
+}
+
 // SetMap sets map for the given key.
 func (s DummyStore) SetMap(key string, value map[string]interface{}) error {
```
Comment (gillesfabio) [inline dummy.go]: The same. `func (DummyStore)`.

File: kvstores_test.go [hunk 3]
```diff
@@ -60,7 +60,17 @@ func testStore(t *testing.T, store KVStore) {
 		exists, err := store.Exists(key)
 		is.Nil(err)
 		is.True(exists)
+	}
+
+	keys := []string{"key1", "key2", "key3"}
 
+	results, err := store.GetMaps(keys)
+
```
Comment (gillesfabio) [inline kvstores_test.go]: Missing `err` check. `is.NoError(err)`.

File: kvstores_test.go [hunk 4]
```diff
@@ -60,7 +60,17 @@ func testStore(t *testing.T, store KVStore) {
...
+	for key, result := range results {
+		is.Equal(result, mapResults[key])
+	}
+
+	for key, _ := range mapResults {
```
Comment (gillesfabio) [inline kvstores_test.go]: Don't need blank. `for key := range mapResults`.

File: kvstores_test.go [hunk 5]
```diff
@@ -60,7 +60,17 @@ func testStore(t *testing.T, store KVStore) {
...
 		err = store.Delete(key)
 		is.Nil(err)
```
Comment (gillesfabio) [inline kvstores_test.go]: Better to use `is.NoError` with errors for consistency.

File: redis.go [hunk 6]
```diff
@@ -108,6 +109,41 @@ func (r *RedisStore) GetMap(key string) (map[string]interface{}, error) {
...
+func (r *RedisStore) GetMaps(keys []string) (map[string]map[string]interface{}, error) {
+	pipe := r.client.Pipeline()
+
+	commands := make(map[string]*redis.StringStringMapCmd, len(keys))
+	for _, key := range keys {
+		commands[key] = pipe.HGetAll(key)
+	}
+
+	_, err := pipe.Exec()
+
```
Comment (gillesfabio) [inline redis.go]: No `\n`. err check must follow err line.

Example:

```go
_, err := pipe.Exec()
if err != nil {
}
```

File: redis.go [hunk 7]
```diff
@@ -289,3 +258,116 @@ func NewRedisClusterStore(options *RedisClusterOptions, expiration time.Duration
...
+// Pipeline uses pipeline as a Redis client to execute multiple calls at once
+func (r *RedisStore) Pipeline(f func(r *RedisStore) error) error {
+	pipe := r.client.Pipeline()
+
+	redisPipeline := RedisPipeline{
+		pipeline: pipe,
+	}
+
+	store := &RedisStore{
+		client:     redisPipeline,
```
Comment (louiseGrandjonc) [inline redis.go:294]: I have the following error : 
```
./redis.go:271: cannot use redisPipeline (type RedisPipeline) as type RedisClient in field value:
	RedisPipeline does not implement RedisClient (Close method has pointer receiver)
```

@gillesfabio @thoas  can you explain me why? :/

**Comment by thoas** [comment]
great stuff 👍

**Comment by louiseGrandjonc** [comment]
thank you :)

**Pull request merged**